src/main/java/org/broadinstitute/hellbender/utils/read/GATKRead.java

```diff
@@ -124,6 +124,51 @@ default boolean isEmpty() {
      */
     int getUnclippedEnd();
 
+    /**
+     * Calculates the reference coordinate for the beginning of the read taking into account soft clips but not hard clips.
+     *
+     * Note: {@link #getUnclippedStart} adds soft and hard clips, this method only adds soft clips.
+     *
+     * @return the unclipped start of the read taking soft clips (but not hard clips) into account
+     */
+    int getSoftStart();
+
+    /**
+     * Calculates the reference coordinate for the end of the read taking into account soft clips but not hard clips.
+     *
+     * Note: {@link #getUnclippedEnd} adds soft and hard clips, this method only adds soft clips.
+     *
+     * @return the unclipped end of the read taking soft clips (but not hard clips) into account
+     */
+    int getSoftEnd();
+
+    /**
+     * Finds the adaptor boundary around the read and returns the first base inside the adaptor that is closest to
+     * the read boundary. If the read is in the positive strand, this is the first base after the end of the
+     * fragment (Picard calls it 'insert'), if the read is in the negative strand, this is the first base before the
+     * beginning of the fragment.
+     *
+     * There are two cases we need to treat here:
+     *
+     * 1) Our read is in the reverse strand :
+     *
+     *     <----------------------| *
+     *   |--------------------->
+     *
+     *   in these cases, the adaptor boundary is at the mate start (minus one)
+     *
+     * 2) Our read is in the forward strand :
+     *
+     *   |---------------------->   *
+     *     <----------------------|
+     *
+     *   in these cases the adaptor boundary is at the start of the read plus the inferred insert size (plus one)
+     *
+     * @return the reference coordinate for the adaptor boundary (effectively the first base IN the adaptor, closest to the read).
+     * CANNOT_COMPUTE_ADAPTOR_BOUNDARY if the read is unmapped or the mate is mapped to another contig.
+     */
+    int getAdaptorBoundary();
```

Comment:
I think this may be problematic to include on this interface because I think it relies on the our definition of fragment size which is neither universally accepted or universally used.

Reply:
It's also super specific to our sequencing technology and just bizarre to have on GATKRead.  I don't like it.

Reply:
Well, it's called `GATKRead` after all, not `Read` :) The benefits of caching this value per read are so huge that I think we have to live with the awkwardness.